packages/eslint-plugin-office-addins/src/rules/load-object-before-read.ts

```diff
@@ -0,0 +1,292 @@
+      "getInvalidCellsOrNullObject",
+      "getDimensionValues",
+      "getByNamespace",
+      "getXml",
+      "getColumnLabelRange",
+      "getDataHierarchy",
+      "getFilterAxisRange",
+      "getPivotItems",
+      "getRowLabelRange",
+      "getDefault",
+      "getPrintArea",
+      "getPrintAreaOrNullObject",
+      "getPrintTitleColumns",
+      "getPrintTitleColumnsOrNullObject",
+      "getPrintTitleRows",
+      "getPrintTitleRowsOrNullObject",
+      "getParentComment",
+      "getAsImage",
+      "getActivePage",
+      "GetStencilInfo",
+      "getBase64Image",
+      "getHtml",
+      "getParagraphInfo",
+      "getByTitle",
+      "getRestApiId",
+      "getByName",
+      "getWindowSize",
+      "getActiveSectionOrNull",
+      "getActiveSection",
+      "getActiveParagraphOrNull",
+      "getActiveParagraph",
+      "getActivePageOrNull",
+      "getActivePage",
+      "getActiveOutlineOrNull",
+      "getActiveOutline",
+      "getActiveNotebookOrNull",
+      "getActiveNotebook",
+      "getCellPadding",
+      "getBorder",
+      "getParagraphBefore",
+      "getParagraphBeforeOrNullObject",
+      "getParagraphAfterOrNullObject",
+      "getParagraphAfter",
+      "getCellOrNullObject",
+      "getHeader",
+      "getFooter",
+      "getTextRanges",
+      "getOoxml",
+      "getNextTextRangeOrNullObject",
+      "getNextTextRange",
+      "getHyperlinkRanges",
+      "getLastOrNullObject",
+      "getDescendants",
+      "getAncestorOrNullObject",
+      "getAncestor",
+      "getByIdOrNullObject",
+      "getById",
+      "getLevelString",
+      "getLevelParagraphs",
+      "getBase64ImageSrc",
+      "getSelection",
+      "getByTypes",
+      "getByTag",
+      "getReferenceId", 
+      "getDocument",
+    ]);
+
+    function getPropertyThatHadToBeLoaded(node: TSESTree.Node | undefined): string | undefined {
+      if(node
+        && node.type === TSESTree.AST_NODE_TYPES.MemberExpression
+        && node.property.type === TSESTree.AST_NODE_TYPES.Identifier) {
+        return node.property.name;
+      }
+      return undefined;
+    }
+
+    function isLoadFunction(node: TSESTree.MemberExpression): boolean {
+      return (node.property.type === TSESTree.AST_NODE_TYPES.Identifier
+        && node.property.name === "load");
+    }
+
+    function callsGetAPIFunction(node: TSESTree.Identifier): boolean {
+      return getFunctions.has(node.name);
+    }
+
+    function isGetVariableDeclaration(node: TSESTree.VariableDeclarator): boolean {
+      if(node.init?.type === TSESTree.AST_NODE_TYPES.CallExpression
+        && node.init.callee.type === TSESTree.AST_NODE_TYPES.MemberExpression
+        && node.init.callee.property.type === TSESTree.AST_NODE_TYPES.Identifier) {
+          if(callsGetAPIFunction(node.init.callee.property)) {
+            return true;
+          }
+      }
+      return false;
+    }
+
+    function isGetAssignmentExpression(node: TSESTree.AssignmentExpression): boolean {
```

Comment:
This looks almost identical to isGetVariableDeclaration.  If you use the CallExpression node as the method argument then these could be collapsed into one.

Reply:
I merged both functions but I didn't quite figure how to merge the two ifs pairs easily.
I also made the statements inside the if much easier to read